src/Microsoft.OData.Core/HttpHeaderValueLexer.cs

```diff
@@ -256,15 +245,15 @@ private static KeyValuePair<string, string> ReadKeyValuePair(ref HttpHeaderValue
                     lexer.Type == HttpHeaderValueItemType.Token || lexer.Type == HttpHeaderValueItemType.QuotedString,
                     "lexer.Type == HttpHeaderValueItemType.Token || lexer.Type == HttpHeaderValueItemType.QuotedString");
 
-                value = lexer.OriginalText;
+                value = lexer.OriginalText.ToString();

```

Comment:
Can `value` be of type `ReadOnlyMemory<char>` instead of `string` to avoid calling `ToString`?

Reply:
Probably this line https://github.com/OData/odata.net/pull/3097/files#diff-d4037d6fdf2743a1c6adbc34ea8ea80637de11e94ed36d4ab9e3250b396a3981R256 means there might be no value in doing that...

Reply:
it definitely needs to be a string to the KeyValuePair. So, ToString() is necessary and You may notice it only happens once